proxy/proxy.go

```diff
@@ -375,21 +376,23 @@ func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	start = time.Now()
-	response := c.Response()
 	if p.preserveOriginal {
-		c.originalResponse = cloneResponseMetadata(response)
+		c.originalResponse = cloneResponseMetadata(c.Response())
 	}
 	p.applyFiltersToResponse(processedFilters, c)
 	metrics.MeasureAllFiltersResponse(rt.Id, start)
 
-	start = time.Now()
-	addBranding(response.Header)
-	copyHeader(w.Header(), response.Header)
-	w.WriteHeader(response.StatusCode)
-	err := copyStream(w.(flusherWriter), response.Body)
-	if err != nil {
-		log.Error(err)
-	} else {
-		metrics.MeasureResponse(response.StatusCode, r.Method, rt.Id, start)
+	if !c.served {
```

Comment:
this is to fully support the deprecated functionality without breaking old stuff